x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_events_query.ts

```diff
@@ -15,8 +15,8 @@ interface BuildEventsSearchQuery {
   to: string;
   filter: estypes.QueryDslQueryContainer;
   size: number;
-  sortOrder?: estypes.SearchSortOrder;
-  searchAfterSortIds: estypes.SearchSortResults | undefined;
+  sortOrder?: estypes.SortOrder;
+  searchAfterSortIds: estypes.SortResults | undefined;
```

Comment:
@ashokaditya Happy to accept those changes. However, IMO, they are not exactly the same:
AFAIK, `searchAfterSortIds: estypes.SortResults | undefined;` enforces the property to be present but it accepts undefined. Setting a property as optional means that the property might not be provided.

It's a small difference, but depending on where it's used, it can catch some errors.

If you're happy with making the property optional, I'll gladly accept your suggestion :)

Reply:
@afharo yes quite right. At least in these cases, the types look like they are optional. Although it does make sense to explicitly define this so that the param is explicitly assigned `undefined` when that method is invoked if that is what is intended 👍